src/main/java/de/tum/in/www1/artemis/service/AssessmentService.java

```diff
@@ -33,23 +34,35 @@
 
     protected final ResultService resultService;
 
+    private final ExamService examService;
+
     private final SubmissionRepository submissionRepository;
 
     public AssessmentService(ComplaintResponseService complaintResponseService, ComplaintRepository complaintRepository, FeedbackRepository feedbackRepository,
             ResultRepository resultRepository, StudentParticipationRepository studentParticipationRepository, ResultService resultService,
-            SubmissionRepository submissionRepository) {
+            SubmissionRepository submissionRepository, ExamService examService) {
         this.complaintResponseService = complaintResponseService;
         this.complaintRepository = complaintRepository;
         this.feedbackRepository = feedbackRepository;
         this.resultRepository = resultRepository;
         this.studentParticipationRepository = studentParticipationRepository;
         this.resultService = resultService;
         this.submissionRepository = submissionRepository;
+        this.examService = examService;
     }
 
     Result submitResult(Result result, Exercise exercise, Double calculatedScore) {
         Double maxScore = exercise.getMaxScore();
-        result.setRatedIfNotExceeded(exercise.getDueDate(), result.getSubmission().getSubmissionDate());
+
+        if (exercise.hasExerciseGroup()) {
+            Exam exam = exercise.getExerciseGroup().getExam();
+            ZonedDateTime latestIndividualExamEndDate = examService.getLatestIndiviudalExamEndDate(exam);
```

Comment:
I'm not sure, but should this really depend on the last individual end date and not on the one belonging to the StudentExam to which the exam belongs?

Reply:
I was unsure here too, maybe @krusche can help

Reply:
Optimally, this should use the individual due date. I would consider to add an appropriate method to `ExamService` that returns the individual due date for a student based on: [`StudentExamService.java` Line 112](https://github.com/ls1intum/Artemis/blob/develop/src/main/java/de/tum/in/www1/artemis/service/StudentExamService.java#L112)

Reply:
Theoretically you are right and this should use the individual date.

However, I think the exam correction should NOT start before the last student exam is over.
So tutors should not even be able to navigate into the tutor exam dashboard in this case and then this check is fine as well

Reply:
We disabled the tutor exam dashboard button until the last individual exam end date is reached. So this should be fine :)